packages/core/lib/dlc/PolynomialPayoutCurve.ts

```diff
@@ -0,0 +1,187 @@
+   */
+  getPayout(outcome: bigint): BigNumber {
+    const { left, slope } = this;
+
+    const x = new BigNumber(Number(outcome));
+
+    // y = mx + b
+    const payout = slope.times(x.minus(left.outcome)).plus(left.payout);
+
+    return payout;
+  }
+
+  /**
+   * Get the outcome for a given payout
+   * @param payout The payout to get the outcome for
+   * @returns The outcome for the payout
+   */
+  getOutcomeForPayout(payout: BigNumber): bigint {
+    const { left, slope } = this;
+    const y = new BigNumber(Number(payout));
+
+    // Find the x value for the given y
+    // slope = (y2 - y1) / (x2 - x1)
+    // x1 = (y2 - y1) / slope + x2
+    const outcome = y
+      .minus(left.payout)
+      .dividedBy(slope)
+      .plus(left.outcome)
+      .integerValue();
+    return BigInt(outcome.toString());
+  }
+
+  /**
+   * Serializes PolynomialPayoutCurve to a PolynomialPayoutCurvePiece (for transport)
+   * @returns A PolynomialPayoutCurvePiece
+   */
+  toPayoutCurvePiece(): PolynomialPayoutCurvePiece {
+    const { points } = this;
+
+    const piece = new PolynomialPayoutCurvePiece();
+
+    piece.points = points.map((point) => {
+      const eventOutcome = BigInt(point.outcome.toString());
+      const outcomePayout = BigInt(point.payout.toString());
+      const extraPrecision = getPrecision(point.payout);
+      return { eventOutcome, outcomePayout, extraPrecision };
+    });
+
+    return piece;
+  }
+
+  /**
+   * Determine if the payout curve is equal to another
+   * @param curve A PolynomialPayoutCurve
+   * @returns True if the curves are the same
+   */
+  equals(curve: PolynomialPayoutCurve): boolean {
+    if (this.points.length !== curve.points.length) return false;
+
+    return this.points.every((point, i) => {
+      const otherPoint = curve.points[i];
+      return (
+        point.outcome.eq(otherPoint.outcome) &&
+        point.payout.eq(otherPoint.payout)
+      );
+    });
+  }
+
+  /**
+   * Creates a PolynomialPayoutCurve from a PolynomialPayoutCurvePiece
+   * @param piece
+   * @returns A PolynomialPayoutCurve
+   */
+  static fromPayoutCurvePiece(
+    piece: PolynomialPayoutCurvePiece,
+  ): PolynomialPayoutCurve {
+    const points = piece.points.map((point) => {
+      const outcome = new BigNumber(point.eventOutcome.toString());
+      const payout = new BigNumber(point.outcomePayout.toString()).plus(
+        fromPrecision(point.extraPrecision),
+      );
+      return { outcome, payout };
+    });
+    return new PolynomialPayoutCurve(points);
+  }
+
+  /**
+   * Computes all CETs for a given payout curve
+   * @param payoutFunction The payout function
+   * @param totalCollateral The total collateral
+   * @param roundingIntervals The rounding intervals
+   * @returns A list of CETs
+   */
+  static computePayouts(
+    payoutFunction: PayoutFunctionV0,
+    totalCollateral: bigint,
+    roundingIntervals: RoundingIntervalsV0,
+  ): CETPayout[] {
+    if (payoutFunction.pieces.length !== 3)
```

Comment:
What about in the case where there's a single line from (0, max_outcome)?
You would only require one curve piece.

Other cases:
- max loss defined, but not max gain    _/
- max gain, but no max loss   /‾‾

Ideally, this function should accept any number (>0) of curve pieces.
For our case, check if `splitIntoRanges` supports a flat line and simply outputs one payout cet payout:

```
[{
         payout: payoutFunction.pieces[0].endpointPayout,
         indexFrom: payoutFunction.endpoint0,
         indexTo: payoutFunction.pieces[0].endpoint,
       }]
```

if so, we can just map `splitIntoRanges` over the provided curve pieces